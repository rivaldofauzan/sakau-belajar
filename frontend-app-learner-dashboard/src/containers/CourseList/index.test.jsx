import { shallow } from '@edx/react-unit-test-utils';

import { reduxHooks } from 'hooks';
import { useCourseListData, useIsCollapsed } from './hooks';
import CourseList from '.';

jest.mock('hooks', () => ({
  reduxHooks: { useHasCourses: jest.fn() },
}));

jest.mock('./hooks', () => ({
  useCourseListData: jest.fn(),
  useIsCollapsed: jest.fn(),
}));

jest.mock('containers/CourseCard', () => 'CourseCard');
jest.mock('containers/CourseFilterControls', () => ({
  ActiveCourseFilters: 'ActiveCourseFilters',
  CourseFilterControls: 'CourseFilterControls',
}));

reduxHooks.useHasCourses.mockReturnValue(true);

describe('CourseList', () => {
  const defaultCourseListData = {
    filterOptions: {},
    numPages: 1,
    setPageNumber: jest.fn().mockName('setPageNumber'),
    showFilters: false,
    visibleList: [],
  };
  useIsCollapsed.mockReturnValue(false);
  const createWrapper = (courseListData) => {
    useCourseListData.mockReturnValueOnce({
      ...defaultCourseListData,
      ...courseListData,
    });
    return shallow(<CourseList />);
  };

  describe('no courses', () => {
    test('snapshot', () => {
      reduxHooks.useHasCourses.mockReturnValue(true);
      const wrapper = createWrapper();
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
  describe('no filters', () => {
    test('snapshot', () => {
      const wrapper = createWrapper();
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
  describe('with filters', () => {
    test('snapshot', () => {
      const wrapper = createWrapper({
        filterOptions: { abitary: 'filter' },
        showFilters: true,
      });
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
  describe('with multiple courses and pages', () => {
    test('snapshot', () => {
      const wrapper = createWrapper({
        visibleList: [{ cardId: 'foo' }, { cardId: 'bar' }, { cardId: 'baz' }],
        numPages: 3,
      });
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
  describe('collapsed with multiple courses and pages', () => {
    test('snapshot', () => {
      useIsCollapsed.mockReturnValueOnce(true);
      const wrapper = createWrapper({
        visibleList: [{ cardId: 'foo' }, { cardId: 'bar' }, { cardId: 'baz' }],
        numPages: 3,
        showFilters: true,
      });
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
});
