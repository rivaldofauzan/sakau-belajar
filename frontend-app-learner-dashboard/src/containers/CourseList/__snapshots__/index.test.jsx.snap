// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList collapsed with multiple courses and pages snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <Fragment>
    <div
      id="course-list-active-filters-container"
    >
      <ActiveCourseFilters />
    </div>
    <div
      className="d-flex flex-column flex-grow-1"
    >
      <CourseCard
        cardId="foo"
        key="foo"
      />
      <CourseCard
        cardId="bar"
        key="bar"
      />
      <CourseCard
        cardId="baz"
        key="baz"
      />
      <Pagination
        className="mx-auto mb-2"
        onPageSelect={[MockFunction setPageNumber]}
        pageCount={3}
        paginationLabel="Course List"
        variant="reduced"
      />
    </div>
  </Fragment>
</div>
`;

exports[`CourseList no courses snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <Fragment>
    <div
      className="d-flex flex-column flex-grow-1"
    />
  </Fragment>
</div>
`;

exports[`CourseList no filters snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <Fragment>
    <div
      className="d-flex flex-column flex-grow-1"
    />
  </Fragment>
</div>
`;

exports[`CourseList with filters snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls
        abitary="filter"
      />
    </div>
  </div>
  <Fragment>
    <div
      id="course-list-active-filters-container"
    >
      <ActiveCourseFilters
        abitary="filter"
      />
    </div>
    <div
      className="d-flex flex-column flex-grow-1"
    />
  </Fragment>
</div>
`;

exports[`CourseList with multiple courses and pages snapshot 1`] = `
<div
  className="course-list-container"
>
  <div
    className="course-list-heading-container"
  >
    <h2
      className="course-list-title"
    >
      My Courses
    </h2>
    <div
      className="course-filter-controls-container"
    >
      <CourseFilterControls />
    </div>
  </div>
  <Fragment>
    <div
      className="d-flex flex-column flex-grow-1"
    >
      <CourseCard
        cardId="foo"
        key="foo"
      />
      <CourseCard
        cardId="bar"
        key="bar"
      />
      <CourseCard
        cardId="baz"
        key="baz"
      />
      <Pagination
        className="mx-auto mb-2"
        onPageSelect={[MockFunction setPageNumber]}
        pageCount={3}
        paginationLabel="Course List"
        variant="secondary"
      />
    </div>
  </Fragment>
</div>
`;
